pkg/kv/kvserver/replica_init.go

```diff
@@ -396,7 +396,7 @@ func (r *Replica) setDescLockedRaftMuLocked(ctx context.Context, desc *roachpb.R
 
 	r.rangeStr.store(r.replicaID, desc)
 	r.isInitialized.Set(desc.IsInitialized())
-	r.connectionClass.set(rpc.ConnectionClassForKey(desc.StartKey))
+	r.connectionClass.set(rpc.ConnectionClassForKey(desc.StartKey, rpc.RaftClass))
```

Comment:
@erikgrinaker Looks good? Should we add a cluster setting to flip this back to `DefaultClass` if needed?

Reply:
Yeah, seems fine. We could consider something like `rpc.ShouldUseSystemConnectionClass()` instead, but I guess it's just going to be more verbose.

>Should we add a cluster setting to flip this back to DefaultClass if needed?

Nah, I doubt this is going to cause any problems. We can add an envvar if we think we need one, but I don't see why we would.

Reply:
> We could consider something like rpc.ShouldUseSystemConnectionClass() instead, but I guess it's just going to be more verbose.

This seems ok, as we already have [some code](https://github.com/cockroachdb/cockroach/blob/9530d1adb5ae5c6136455178617c56201deb443b/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go#L959-L964) for figuring out the class outside the `rpc` package. So could have more.

I feel like `rpc.ConnectionClassForKey` is a bit out of place. Seemingly, the `rpc` package shouldn't be aware of the keys layout. So, arguably, this code should move to upper layers anyway, and just tell the `rpc` package which class to use.

Reply:
🤷 I could see us having a generalized mechanism for marking system ranges, but it isn't clear that we'd want all system ranges to use the system class -- for example, consider the time series ranges or the `system` database (which contains a ton of observability data).

It seems fine for now to keep something like this in the `rpc` package, and limit the scope specifically to RPC class conditions, but I don't have a strong opinion on how to structure it. We can maybe revisit this once we've implemented all the classes and have a better sense of what's needed.

Reply:
Keeping this simple for now. Added an env var to have a backstop. Made it static to a) avoid the overhead of reading the env var in every `Replica` init path, b) avoid the "eventually consistent" process state in which we use both `DefaultClass` and `RaftClass` while switching between the two, and have to garbage collect one.

Let's get back to system ranges aspects in #116958.